### ISSUE TITLE
Unable to commit in pre-commit enabled repository

### ISSUE THREAD

**Issue by gegoune** [TimUntersberger/neogit#351]
### Description

I am unable to make commits for a little while now, last working version on master is 64245bb, after that got merged committing in repository with configured pre-commit fails. I do have some unstaged changes and command line `git commit` prints out coloured output from pre-commit but takes me to nvim to write message and everything works.

```
 git --no-optional-locks commit
[WARNING] Unstaged files detected.
[INFO] Stashing unstaged files to /Users/gegoune/.cache/pre-commit/patch1664292138-36871.
StyLua (Github)..........................................................Passed
[INFO] Restored changes from /Users/gegoune/.cache/pre-commit/patch1664292138-36871.
[main somesha] some prev commit
 1 file changed, 1 insertion(+), 1 deletion(-)
```

`Warning` and `Passed` are coloured.

### Neovim version

NVIM v0.8.0-dev-2635-g1d337d4e2-dirty
Build type: Release
LuaJIT 2.1.0-beta3

### Operating system and version

macOS 12.6

### Steps to reproduce

1. Create repository with pre-commit configuration and some unstated changes
2. `nvim -nu minimal.lua`
3. `:Neogit`, stage one of the changes leaving some (this is what I have in my repo that fails) unstaged, press `c`, `c`
4. Type commit message and `:wq`

### Expected behavior

Create commit without erroring.

### Actual behavior

Neogit fails to create commit, erroring with:

```
Git Error (1), press $ to see the git command history
```
`$` shows failed command:

```1 git --no-optional-locks commit (0.090 ms) [stderr 5]```

### Minimal config

```Lua
vim.cmd([[set runtimepath=$VIMRUNTIME]])
vim.cmd([[set packpath=/tmp/nvim/site]])
local package_root = "/tmp/nvim/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup({
    {
      "wbthomason/packer.nvim",
      {
        "TimUntersberger/neogit",
        requires = {
          { "nvim-lua/plenary.nvim" },
          { "sindrets/diffview.nvim" },
        },
        config = function()
          print("loaded neogit")
          require("neogit").setup()
        end,
      },
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  })
end
if vim.fn.isdirectory(install_path) == 0 then
  print("Installing neogit and dependencies.")
  vim.fn.system({ "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path })
end
load_plugins()
require("packer").sync()
```

**Comment by gegoune** [comment]
Unfortunately log does not help:

```
[ERROR <time>] ...site/pack/packer/start/neogit/lua/neogit/lib/git/cli.lua:336: Execution of 'git --no-optional-locks commit'
[ERROR <time>] ...site/pack/packer/start/neogit/lua/neogit/lib/git/cli.lua:338:   failed with code 1
[ERROR <time>] ...site/pack/packer/start/neogit/lua/neogit/lib/git/cli.lua:340:   took 1 ms
[DEBUG <time>] ...nvim/site/pack/packer/start/neogit/lua/neogit/status.lua:329: [STATUS BUFFER]: Starting refresh
```

**Comment by gegoune** [comment]
Fixed by #355
